overlord/state/task.go

```diff
@@ -95,6 +106,41 @@ func (t *Task) Summary() string {
 	return t.summary
 }
 
+// Status returns the current task status.
+func (t *Task) Status() Status {
+	t.state.ensureLocked()
+	return t.status
+}
+
+// SetStatus sets the task status, overriding the default behavior (see Status method).
+func (t *Task) SetStatus(s Status) {
+	t.state.ensureLocked()
+	t.status = s
+}
+
+// Progress returns the current progress for the task.
+// If progress is not explicitly set, it returns (0, 1) if the status is
+// Running or Waiting and (1, 1) otherwise.
+func (t *Task) Progress() (cur, total int) {
+	t.state.ensureLocked()
+	if t.curProgress == 0 && t.totProgress == 0 {
```

Comment:
We can test the total value only here. It'd be an error to have cur > 0 if total is 0, so we can fallback to the default behavior.